src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilterTest.java

```diff
@@ -194,82 +171,35 @@ private static SuppressionXpathFilter createSuppressionXpathFilter(String fileNa
-                .isFalse();
-
-        assertWithMessage("TreeWalker audit event should be rejected")
-                .that(filter.accept(createTreeWalkerAudit(24, 28, TokenTypes.STRING_LITERAL, file)))
-                .isFalse();
+        final String[] expected = {
+            "17:29: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+            "19:29: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+            "21:29: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+        };
+
+        final String[] suppressed = {
+            "17:29: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+            "21:29: " + getCheckMessage(AvoidEscapedUnicodeCharactersCheck.class, MSG_KEY),
+        };
+
+        verifyFilterWithInlineConfigParser(getPath("InputSuppressionXpathFilterEscapeString.java"),
+                                           expected, removeSuppressed(expected, suppressed));
     }
 
     @Test
     public void testFalseEncodeChar() throws Exception {
-        final boolean optional = false;
-        final SuppressionXpathFilter filter = createSuppressionXpathFilter(
-                getPath("InputSuppressionXpathFilterEscapeChar.xml"), optional);
-        final File file = new File(getPath("InputSuppressionXpathFilterEscapeChar.java"));
-
-        assertWithMessage("TreeWalker audit event should be rejected")
-                .that(filter.accept(createTreeWalkerAudit(8, 13, TokenTypes.CHAR_LITERAL, file)))
-                .isFalse();
-
-        assertWithMessage("TreeWalker audit event should be rejected")
-                .that(filter.accept(createTreeWalkerAudit(10, 13, TokenTypes.CHAR_LITERAL, file)))
-                .isFalse();
-
-        assertWithMessage("TreeWalker audit event should be rejected")
-                .that(filter.accept(createTreeWalkerAudit(12, 13, TokenTypes.CHAR_LITERAL, file)))
-                .isFalse();
-
-        assertWithMessage("TreeWalker audit event should be rejected")
-                .that(filter.accept(createTreeWalkerAudit(14, 13, TokenTypes.CHAR_LITERAL, file)))
-                .isFalse();
-
-        assertWithMessage("TreeWalker audit event should be rejected")
-                .that(filter.accept(createTreeWalkerAudit(16, 13, TokenTypes.CHAR_LITERAL, file)))
-                .isFalse();
-    }
-
-    private static TreeWalkerAuditEvent createTreeWalkerAudit(int lineNo, int columnNo,
-                                                              int tokenTypes, File file)
-            throws IOException, CheckstyleException {
-        final Violation message = new Violation(lineNo, columnNo, tokenTypes, "", "",
-                null, null, "777", SuppressionXpathFilterTest.class, null);
-        return new TreeWalkerAuditEvent(null, "Test.java", message,
-                JavaParser.parseFile(file, JavaParser.Options.WITHOUT_COMMENTS));
```

Comment:
@romani The reason for doing those changes is that till now this test was hypothetical, it was itself creating some custom violations (not based on any check AFAIK) and then creating suppression for it and suppressing those violations, the test cases now are based on a real check, since the hypothetical violations were about string literals, I used AvoidEscpaedUnicode so the flavour of suppression remained same.

Reply:
I know